src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,651 @@ public ByteArraySet createSmallSet() {
+    }
+
+    @Test
+    public void bigIterator() {
+        testIterator(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigIterator() {
+        testIterator(SIZE_REALLY_BIG);
+    }
+
+    private void testRemove(int size) {
+        System.out.println("Testing remove method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        if (size > 0) {
+            assertFalse(set.isEmpty());
+        } else {
+            assertTrue(set.isEmpty());
+        }
+        Collections.shuffle(vals);
+
+        int count = 0;
+        for(ByteArray val: set) {
+            assertTrue(vals.contains(val));
+            count++;
+        }
+        assertEquals(size, count);
+
+        int currentSize = size;
+        for(ByteArray val: vals) {
+            assertTrue(set.remove(val));
+            assertFalse(set.contains(val));
+            currentSize -= 1;
+            assertEquals(currentSize, set.size());
+        }
+    }
+
+    @Test
+    public void emptyRemove() {
+        testRemove(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallRemove() {
+        testRemove(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallRemove() {
+        testRemove(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularRemove() {
+        testRemove(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigRemove() {
+        testRemove(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigRemove() {
+        testRemove(SIZE_REALLY_BIG);
+    }
+
+    private void testRandomSerializeDerializeSizeCheckRemove(int size) {
+        System.out.println("Testing serialize/deserialize methods with random-size check with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> originalVals = getRandomByteArrays(size);
+
+        List<ByteArray> vals = new ArrayList<>();
+        for (ByteArray val: originalVals) {
+            vals.add(val);
+        }
+        Collections.shuffle(vals);
+
+        int forceMergerSize = (new Random()).nextInt(size);
+        List<String> insertedStringVals = new ArrayList<>();
+        for (ByteArray val: vals) {
+            insertedStringVals.add(new String(val.getBytes()));
+            assertTrue(set.add(val));
+            if (set.size() == forceMergerSize) {
+                // Forces the merger of the fronting set into the list of chunks
+                set.serialize();
+            }
+        }
+
+        int deserializedCheckSize = (new Random()).nextInt(size);
+        int currentSize = size;
+        List<String> deletedStringVals = new ArrayList<>();
+        for (ByteArray val: originalVals) {
+            if (currentSize == deserializedCheckSize) {
+                if (currentSize != ByteArraySet.deserialize(set.serialize()).size()) {
+                    assertEquals("",
```

Comment:
This assertion looks wrong maybe? Looks to be comparing equals of an empty string on a string formatted value. I don't see this ever passing if this condition gets hit